snuba/datasets/events/split.py

```diff
@@ -0,0 +1,169 @@
+import copy
```

Comment:
The logic in this file is the same as before.

I just moved the `_time_split` and `_col_split` functions out of the wrapper cause they don't really need to be wrapped anymore.

I have intentionally not tried to make classes and strategies out of this because we don't know what kind of performance tweeks the other datasets will need, so, don't want to over-engineer it.